tools/ci_build/build.py

```diff
@@ -163,7 +164,7 @@ def convert_arg_line_to_args(self, arg_line):
         "--parallel",
         nargs="?",
         const="0",
-        default="1",
+        default=str(cpu_count()),
```

Comment:
using a value of 0 will have a similar effect:
https://github.com/microsoft/onnxruntime/blob/2de883c5923a1c3cd4d72a5f69a708595bd81c2a/tools/ci_build/build.py#L2723

also maybe update this option's help text to indicate that builds will be parallel by default.